### PR TITLE
[BEAM-3294] Make Go SDK external transform a primitive

### PR DESCRIPTION
**Pull request by herohde**
This change allows the removal of the last source/sink remnants. Also included minor cleanup.

### CODE REVIEW

**Comment by herohde** [comment]
R: @wcn3

File: sdks/go/pkg/beam/core/runtime/graphx/serialize.go [hunk 1]
```diff
@@ -318,6 +318,14 @@ func encodeType(t reflect.Type) (*v1.Type, error) {
 	if s, ok := tryEncodeSpecial(t); ok {
 		return &v1.Type{Kind: v1.Type_SPECIAL, Special: s}, nil
 	}
+	if k, ok := runtime.TypeKey(t); ok {
```
Comment (wcn3) [inline sdks/go/pkg/beam/core/runtime/graphx/serialize.go]: Interesting. So this gives users the power now to serialize something like a channel, as long as they register it and do something sane with it.

Comment (herohde) [reply sdks/go/pkg/beam/core/runtime/graphx/serialize.go]: The registered type still has to be a named type, but, yes, users can serialize the type of interfaces and whatnot using graphx. I think the old chan stuff was already there. However, we'd still need a coder registry that can code such types for them to appear in a user fn signature.

**Comment by herohde** [comment]
Thanks!

**Comment by herohde** [comment]
R: @kennknowles (merge)

**Comment by herohde** [comment]
(rebased)

**Pull request merged**